File: 0015-3sum/0015-3sum.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        int n = nums.length;
        Arrays.sort(nums);
        HashSet<List<Integer>> set = new HashSet<>();
        for(int i =0 ;i<n; i++)
        {
            int j = i+1;
            int k = n-1;
            while(j<k)
            {
                List<Integer> triplet = new ArrayList<>();
                int s = nums[i] + nums[j] + nums[k];
                if(s==0)
                {
                    triplet.add(nums[i]);
                    triplet.add(nums[j]);
                    triplet.add(nums[k]);
                    set.add(triplet);
                    j++;
                    k--;
                }
                else if (s>0)
                k--;
                else
                j++;
            }
        }
        List<List<Integer>> ans = new ArrayList<>(set);
        return ans;
    }
}

File: 0015-3sum/0015-3sum.py
class Solution:
    def threeSum(self, nums: list[int]) -> list[list[int]]:
        length = len(nums)
        ans = []
        nums.sort()
        for i in range(0, length):
            if (i>0 and nums[i]==nums[i-1]):
                continue
            left = i+1
            right = length -1
            while(left<right):
                s = nums[left] + nums[right] + nums[i]
                if (s == 0):
                    ans.append([nums[i], nums[left], nums[right]])
                    left+=1
                    right-=1
                    while nums[left] == nums[left-1] and left<right:
                        left+=1
                elif (s < 0):
                    left+=1
                else:
                    right-=1
        return ans